app/src/main/java/com/asfoundation/wallet/rating/RatingInteractor.kt

```diff
@@ -0,0 +1,63 @@
+package com.asfoundation.wallet.rating
+
+import com.appcoins.wallet.bdsbilling.WalletService
+import com.asfoundation.wallet.transactions.Transaction
+import com.asfoundation.wallet.ui.gamification.GamificationInteractor
+import io.reactivex.Completable
+import io.reactivex.Scheduler
+import io.reactivex.Single
+
+class RatingInteractor(private val ratingRepository: RatingRepository,
+                       private val gamificationInteractor: GamificationInteractor,
+                       private val walletService: WalletService,
+                       private val ioScheduler: Scheduler) {
+
+  fun shouldOpenRatingDialog(): Single<Boolean> {
+    val remindMeLaterDate = ratingRepository.getRemindMeLaterDate()
+    if (remindMeLaterDate > -1F && remindMeLaterDate <= System.currentTimeMillis()) {
+      return Single.just(true)
+    }
+    if (!ratingRepository.hasSeenDialog()) {
+      return gamificationInteractor.getUserStats()
+          .map { stats ->
+            if (stats.level >= 6) {
+              true
+            } else {
+              ratingRepository.getSuccessfulTransactions() >= 5
+            }
+          }
+    }
+    return Single.just(false)
+  }
+
+  fun saveTransactionsNumber(transactions: List<Transaction>) {
+    var transactionsNumber = 0L
+    for (transaction in transactions) {
+      if ((transaction.type == Transaction.TransactionType.IAP
+              || transaction.type == Transaction.TransactionType.TOP_UP)
+          && transaction.status == Transaction.TransactionStatus.SUCCESS) {
+        transactionsNumber++
+      }
+    }
+    ratingRepository.saveSuccessfulTransactions(transactionsNumber)
+  }
```

Comment:
I believe this can be optimized in two ways.
1- The for should end when transactionsNumber is over 5
2- Maybe we should only check for the number of transactions if the value saved is 5 or below

Reply:
I thought about that, but figured it could be useful to save the nr of transactions. But honestly, yeah I don't think it's needed.